controllers/util.go

```diff
@@ -199,6 +199,12 @@ func fetchObjectAndSecret(o opts, stateStoreRef client.ObjectKey, stateStore Sta
 		o.logger.Error(err, "unable to fetch resource", "resourceKind", stateStore.GetObjectKind(), "stateStoreRef", stateStoreRef)
 		return nil, err
 	}
+
+	if stateStore.GetSecretRef() == nil {
+		o.logger.Info("No secret ref found, assuming IAM role should be used", "stateStore", stateStoreRef)
```

Comment:
should this be assumed or should the code verify if the authMethod is `iam`?

Reply:
oops, we should just log nothing and continue